### ISSUE TITLE
Empty News sitemap

### ISSUE THREAD

**Issue by turnpiece** [sybrew/The-SEO-Framework-Extension-Manager#41]
I've activated the Articles extension, configured it for posts and ticked the Google News Sitemap box but the sitemap at sitemap-news.xml is empty, and is still empty after waiting for some time, despite the site having numerous posts.

Not sure if this is related, but when I click save on the Extension Settings page I get "Error code: 18105 — An informative notice should have been placed here for the error code, but the server experienced an error."

**Comment by sybrew** [comment]
Hello!

The sitemap should fill immediately. When it doesn't, either a caching plugin is interfering (unlikely), or no posts have been marked as "News Article" via the "Structure -> Article Type" post-SEO-settings.

`18105` means that the extension settings are saved correctly. There's probably another plugin ruining the AJAX response data due to novice programming, so you get that generic fallback message. See if deactivating some plugins change the notice text to "Changes are saved for one extension.", that'll help deduct which plugin is interfering. A theme may also be responsible, of course.

**Comment by turnpiece** [comment]
Thanks. The sitemap was showing yesterday and had articles in it, but today I'm getting error code 2001 and 6001, both saying 

"An error occured while verifying the options. Security keys have likely changed and enabled extensions are now inactive. If this error keeps coming back, please disconnect your account and try again."

That error does keep coming back, and I do keep disconnecting the account and reconnecting it.

**Comment by sybrew** [comment]
Hello!

Do you have a WordPress security key (AUTH_KEY, NONCE_KEY, etc.) scrambler enabled, which reconfigures the keys in your wp-config.php file periodically?

**Comment by turnpiece** [comment]
No, I don't think the security keys have been changed, but we are blocking access to /wp-admin from external IP addresses. Could that be the cause of this?

**Comment by sybrew** [comment]
Hello!

No, that can't be the issue. Next to being unrelated, to clarify the privacy concern that's bound to your assertion: Extension Manager only communicates to us to retrieve responses, we don't communicate the other way round &mdash; the sole exception is our Monitor service, which can crawl your website, but that service cannot affect your website. None of our services can remotely affect your website.

Another cause for 2001/6001 is that the site migrates its database, but not the files, from production to staging and vice versa. In that case it's best to match the keys defined in `wp-config.php` on both sites.

**Comment by jwholdsworth** [comment]
Hello!

I've been working with @turnpiece on this recently, and we're running a multi-server setup. We updated the license key through the Wordpress GUI and wenoticed that the sitemap is now produced correctly on one server, but not on any of the others. Do we need to define `TSF_EXTENSION_MANAGER_API_INFORMATION` with our username and key in wp-config.php (or somewhere else) to ensure consistency across all servers?

**Comment by sybrew** [comment]
Hi @jwholdsworth; yes, if all the wp-config.php salts/hashes of the Beanstalk are synchronized, you can best use that constant definition in wp-config.php or a mu-plugin. The API should then work without issue. If you wish to discuss things more privately, please refer [to our contact page](https://tsf.fyi/contact).

**Comment by sybrew** [comment]
To add to this, we want to robustly support complex servers like yours and staging/production environments. A solution to these issues is in the works, but it might take a while. Sorry for the inconvenience!
